conan/tools/gnu/autotools.py

```diff
@@ -63,11 +63,38 @@ def make(self, target=None):
         command = join_arguments([make_program, target, str_args, jobs])
         self._conanfile.run(command)
 
+    @staticmethod
+    def _collect_shared_lib_files(folder):
```

Comment:
As it collects only for Macos, probably call it ``_osx_collect...`` or something like that?
As it can probably be shortened to:

```python
def _collect_lib_files(folder):
    files = [os.path.join(folder, f) for f in os.listdir(folder)]
    return [f for f in files if f.endswith(".dylib") and not os.path.islink(f)]
```
Maybe make it internal to the ``_fix_macos_install_paths``?